flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adapter/DefaultExecutionVertexTest.java

```diff
@@ -0,0 +1,148 @@
+
+	private final int parallelism = 3;
+
+	private List<SchedulingExecutionVertex> schedulingExecutionVertices;
+
+	private List<ExecutionVertex> executionVertices;
+
+	@Before
+	public void setUp() throws Exception {
+		JobVertex[] jobVertices = new JobVertex[2];
+		jobVertices[0] = createNoOpVertex(parallelism);
+		jobVertices[1] = createNoOpVertex(parallelism);
+		jobVertices[1].connectNewDataSetAsInput(jobVertices[0], ALL_TO_ALL, BLOCKING);
+		jobVertices[0].setInputDependencyConstraint(ALL);
+		jobVertices[1].setInputDependencyConstraint(ANY);
+		ExecutionGraph executionGraph = createSimpleTestGraph(
+			new JobID(),
+			taskManagerGateway,
+			triggeredRestartStrategy,
+			jobVertices);
+		ExecutionGraphToSchedulingTopologyAdapter adapter = new ExecutionGraphToSchedulingTopologyAdapter(executionGraph);
+
+		schedulingExecutionVertices = new ArrayList<>();
+		adapter.getVertices().forEach(vertex -> schedulingExecutionVertices.add(vertex));
+		executionVertices = new ArrayList<>();
+		executionGraph.getAllExecutionVertices().forEach(vertex -> executionVertices.add(vertex));
+	}
+
+	@Test
+	public void testGetId() {
+		for (int idx = 0; idx < schedulingExecutionVertices.size(); idx++){
+			assertEquals(schedulingExecutionVertices.get(idx).getId().getJobVertexId(),
+				executionVertices.get(idx).getJobvertexId());
+			assertEquals(schedulingExecutionVertices.get(idx).getId().getSubtaskIndex(),
+				executionVertices.get(idx).getParallelSubtaskIndex());
+		}
+	}
+
+	@Test
+	public void testGetExecutionState() {
+		final ExecutionState[] states = ExecutionState.values();
+		Random random = new Random();
+		for (ExecutionVertex executionVertex: executionVertices) {
+			setVertexState(executionVertex, states[Math.abs(random.nextInt()) % states.length]);
+		}
+
+		for (int idx = 0; idx < schedulingExecutionVertices.size(); idx++) {
+			assertEquals(schedulingExecutionVertices.get(idx).getState(), executionVertices.get(idx).getExecutionState());
+		}
+	}
+
+	@Test
+	public void testGetInputDependencyConstraint() {
+		for (int idx = 0; idx < schedulingExecutionVertices.size(); idx++) {
+			assertEquals(schedulingExecutionVertices.get(idx).getInputDependencyConstraint(),
+				executionVertices.get(idx).getJobVertex().getInputDependencyConstraint());
+		}
+	}
+
+	@Test
+	public void testGetConsumedResultPartitions() {
+		for (int idx = 0; idx < schedulingExecutionVertices.size(); idx++) {
+			Collection<IntermediateResultPartitionID> partitionIds1 =  schedulingExecutionVertices
+				.get(idx).getProducedResultPartitions().stream().map(SchedulingResultPartition::getId)
+				.collect(Collectors.toList());
+			Set<IntermediateResultPartitionID> partitionIds2 = executionVertices
+				.get(idx).getProducedPartitions().keySet();
+			assertThat(partitionIds1, containsInAnyOrder(partitionIds2.toArray()));
+		}
+	}
+
+	@Test
+	public void testGetProducedResultPartitions() {
+		for (int idx = parallelism; idx < schedulingExecutionVertices.size(); idx++) {
```

Comment:
should start at 0?

Reply:
Sorry for the confusion of wrong method name, actually this was testing `ConsumedResultPartitions`, since only consumer jobVertex#2 has `InputEdge`, thus I start from `parallelism`.